### PR TITLE
Add API seeds generator

### PR DESCRIPTION
**Pull request by yndajas**
## Context

The usefulness of the local server has been diminishing due to a lack of seed data

Related API PR: https://github.com/ministryofjustice/hmpps-accredited-programmes-api/pull/187

## Changes in this PR

- Fix factories
- Add more prisoner stubs
- Add API seeds generator

## Release checklist

[Release process documentation](../doc/how-to/perform-a-release.md)

As part of our continuous deployment strategy we must ensure that this work is
ready to be released once merged.

### Pre-merge

- [ ] There are changes required to the Accredited Programmes API for this change to work...
  - [ ] ... and they have been released to production already

### Post-merge

- [ ] This adds, extends or meaningfully modifies a feature...
  - [ ] ... and I have written or updated an end-to-end test for the happy path in the [Accredited Programmes E2E repo](https://github.com/ministryofjustice/hmpps-accredited-programmes-e2e)
- [ ] This makes new expectations of the API...
  - [ ] ... and I have notified the API developer(s) of changes to the contract tests (Pact), or the API is already compliant
- [ ] [Manually approve](../doc/how-to/perform-a-release.md#releasing-to-the-preprod-environment) release to preprod
- [ ] [Manually approve](../doc/how-to/perform-a-release.md#releasing-to-the-production-environment) release to prod

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -60,14 +60,15 @@ There are three user accounts with different roles that can be used when running
...
-| --------- | -------------------------------------- | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
-| Draft     | `c11fab18-dc8d-420c-9c82-d0edd373732d` | Task list: [Refer](http://localhost:3000/refer/referrals/new/c11fab18-dc8d-420c-9c82-d0edd373732d)                                                                                                                               |
-| Submitted | `9d38204c-bfe9-42e6-b751-f46cec067e57` | Personal details: [Refer](http://localhost:3000/refer/referrals/9d38204c-bfe9-42e6-b751-f46cec067e57/personal-details) \| [Assess](http://localhost:3000/assess/referrals/9d38204c-bfe9-42e6-b751-f46cec067e57/personal-details) |
+This SQL can updated by:
+
+1. running `generate-api-seeds`, which will output a new SQL script to the terminal
+1. using this output to update the seeds script on the API side
```
Comment (Gweaton) [inline README.md:70]: Does this mean: we copy and paste the SQL into the API codebase's `R__Seed_Data.sql` if we want to update the seeds on the backend? i.e. what you've done in your API PR?

File: README.md [hunk 2]
```diff
@@ -60,14 +60,15 @@ There are three user accounts with different roles that can be used when running
...
 
-| Status    | ID                                     | Links                                                                                                                                                                                                                            |
-| --------- | -------------------------------------- | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
-| Draft     | `c11fab18-dc8d-420c-9c82-d0edd373732d` | Task list: [Refer](http://localhost:3000/refer/referrals/new/c11fab18-dc8d-420c-9c82-d0edd373732d)                                                                                                                               |
-| Submitted | `9d38204c-bfe9-42e6-b751-f46cec067e57` | Personal details: [Refer](http://localhost:3000/refer/referrals/9d38204c-bfe9-42e6-b751-f46cec067e57/personal-details) \| [Assess](http://localhost:3000/assess/referrals/9d38204c-bfe9-42e6-b751-f46cec067e57/personal-details) |
+This SQL can updated by:
```
[the remark this answers is not in the document]
Comment (yndajas) [reply README.md:67]: When the seeds don't meet our needs

File: README.md (hunk 1 above)
Comment (yndajas) [reply README.md:70]: Ha, I was trying to find words and gave up - will have another think!

File: README.md [hunk 3]
```diff
@@ -60,14 +60,16 @@ There are three user accounts with different roles that can be used when running
...
+
+1. running `generate-api-seeds`, which will output a new SQL script to the terminal
+1. copying this output into the API seed file and opening a pull request with the changes
```
Comment (Gweaton) [inline README.md:70]: shall we put a link in to the API repo's file here, to prevent people needing to dig around in the API code?

**Review by Gweaton — APPROVED** [review]

File: README.md (hunk 3 above)
Comment (yndajas) [reply README.md]: Have added "(linked above)"

**Pull request merged**